### PR TITLE
[FLINK-36405][runtime][security] Fix startup issues on kerberos clusters.

### PR DESCRIPTION
**Pull request by zhengchenyu**
## What is the purpose of the change

Fix startup issues on kerberos clusters

## Brief change log

  - The flink startup user used as the renewer when calling getDelegationToken.
  - Add `org.apache.flink.table.security.token.HiveServer2DelegationTokenIdentifier` to services.

## Verifying this change

This change added tests and can be verified as follows:

  - verified by flink on k8s

## Documentation

For more information see: https://issues.apache.org/jira/browse/FLINK-36405

### CODE REVIEW

**Review by ferenc-csaky — APPROVED** [review]
Change LGTM! cc @gaborgsomogyi

File: flink-connectors/flink-connector-hive/src/main/resources/META-INF/services/org.apache.hadoop.security.token.TokenIdentifier [hunk 1]
```diff
@@ -0,0 +1,16 @@
+# Licensed to the Apache Software Foundation (ASF) under one or more
```
Comment (gaborgsomogyi) [inline flink-connectors/flink-connector-hive/src/main/resources/META-INF/services/org.apache.hadoop.security.token.TokenIdentifier:1]: Why this file needed?

Comment (zhengchenyu) [reply flink-connectors/flink-connector-hive/src/main/resources/META-INF/services/org.apache.hadoop.security.token.TokenIdentifier:1]: After fix renewer user, still throw exception like below:

```
2024-09-26 14:35:07,144 ERROR org.apache.flink.table.security.token.HiveServer2DelegationTokenProvider [] - Failed to obtain delegation token for HiveServer2
java.lang.NullPointerException: null
    at org.apache.flink.table.security.token.HiveServer2DelegationTokenProvider.getTokenRenewalInterval(HiveServer2DelegationTokenProvider.java:202) ~[flink-sql-connector-hive-3.1.3_2.12-1.19.1.jar:1.19.1]
    at org.apache.flink.table.security.token.HiveServer2DelegationTokenProvider.lambda$obtainDelegationTokens$0(HiveServer2DelegationTokenProvider.java:170) ~[flink-sql-connector-hive-3.1.3_2.12-1.19.1.jar:1.19.1]
...
```

 I found [tokenIdentifier](https://github.com/apache/flink/blob/da393c92db814803c9ac96c6cdd55ae444c43689/flink-connectors/flink-connector-hive/src/main/java/org/apache/flink/table/security/token/HiveServer2DelegationTokenProvider.java#L149C70-L149C85) is null. Through further debugging, I found HiveServer2DelegationTokenIdentifier is not in services file, then org.apache.hadoop.security.token.Token::getClassForIdentifier can not recognize the kind `HIVE_DELEGATION_TOKEN`, then hive2Token.decodeIdentifier() will return null.

Comment (gaborgsomogyi) [reply flink-connectors/flink-connector-hive/src/main/resources/META-INF/services/org.apache.hadoop.security.token.TokenIdentifier:1]: Ah yeah, we use hive to renew to get the expiration time.

**Review by gaborgsomogyi — APPROVED** [review]

File: flink-connectors/flink-connector-hive/src/main/resources/META-INF/services/org.apache.hadoop.security.token.TokenIdentifier (hunk 1 above)
Comment (zhengchenyu) [reply flink-connectors/flink-connector-hive/src/main/resources/META-INF/services/org.apache.hadoop.security.token.TokenIdentifier:1]: > Ah yeah, we use hive to renew to get the expiration time.

Well, if hive is used for renew delegation token, this problem will not occur. But we use another user.

**Pull request merged**